src/Compilers/CSharp/Test/Emit3/Semantics/ExtensionTests.cs

```diff
@@ -19099,6 +19099,810 @@ public void AppendFormatted<T>(T hole, int alignment = 0, string format = null)
+            """;
+
+        var exeSource = """
+            "1".M($"");
+            E.M("2", $"");
+            """;
+
+        var expectedOutput = ExecutionConditionUtil.IsCoreClr ? "12" : null;
+        CompileAndVerify([exeSource, src], targetFramework: TargetFramework.Net90, expectedOutput: expectedOutput, verify: Verification.FailsPEVerify);
+
+        var comp1 = CreateCompilation(src, targetFramework: TargetFramework.Net90);
+
+        CompileAndVerify(exeSource, references: [useMetadataRef ? comp1.ToMetadataReference() : comp1.EmitToImageReference()], targetFramework: TargetFramework.Net90, expectedOutput: expectedOutput, verify: Verification.FailsPEVerify);
+    }
+
+    [Theory, WorkItem("https://github.com/dotnet/roslyn/issues/78137")]
+    [CombinatorialData]
+    public void InterpolationHandler_ReceiverParameter_WithConversion_ExtensionParameterWiderThanConstructor(bool useMetadataRef)
+    {
+        var src = """
+            [System.Runtime.CompilerServices.InterpolatedStringHandler]
+            public struct InterpolationHandler
+            {
+
+                public InterpolationHandler(int literalLength, int formattedCount, string s)
+                {
+                    System.Console.WriteLine(s);
+                }
+                public void AppendLiteral(string value) { }
+                public void AppendFormatted<T>(T hole, int alignment = 0, string format = null) => throw null;
+            }
+
+            public static class E
+            {
+                extension(object o)
+                {
+                    public void M([System.Runtime.CompilerServices.InterpolatedStringHandlerArgument("o")] InterpolationHandler h) {}
+                }
+            }
+            """;
+
+        var exeSource = """
+            "1".M($"");
+            E.M("2", $"");
+            """;
+
+        var expectedDiagnostics = new[] {
+            // (1,1): error CS1503: Argument 3: cannot convert from 'object' to 'string'
+            // "1".M($"");
+            Diagnostic(ErrorCode.ERR_BadArgType, @"""1""").WithArguments("3", "object", "string").WithLocation(1, 1),
+            // (2,5): error CS1503: Argument 3: cannot convert from 'object' to 'string'
+            // E.M("2", $"");
+            Diagnostic(ErrorCode.ERR_BadArgType, @"""2""").WithArguments("3", "object", "string").WithLocation(2, 5)
+        };
+        CreateCompilation([exeSource, src], targetFramework: TargetFramework.Net90).VerifyDiagnostics(expectedDiagnostics);
+
+        var comp1 = CreateCompilation(src, targetFramework: TargetFramework.Net90);
+
+        CreateCompilation(exeSource, references: [useMetadataRef ? comp1.ToMetadataReference() : comp1.EmitToImageReference()], targetFramework: TargetFramework.Net90).VerifyDiagnostics(expectedDiagnostics);
+    }
+
+    [Theory, WorkItem("https://github.com/dotnet/roslyn/issues/78137")]
+    [CombinatorialData]
+    public void InterpolationHandler_ReceiverParameter_ByRef(bool useMetadataRef)
```

Comment:
>InterpolationHandler_ReceiverParameter_ByRef [](http://example.com/codeflow?start=16&length=44)

It looks like we are lacking tests for "by value" struct scenarios. In addition to a trivial case, it would be interesting to test a scenario when a handler constructor modifies the receiver by direct access (not through parameter), for example by changing value of the field used as the receiver. Are we going to observe the change in the receiver value inside the extension method?